core/src/main/java/io/questdb/griffin/SqlParser.java

```diff
@@ -825,9 +825,22 @@ private ExecutionModel parseCreateMatView(
             final QueryModel queryModel = parseDml(lexer, null, lexer.getPosition(), true, sqlParserCallback, null);
             final int endOfQuery = lexer.getPosition() - 1;
 
+            // Basic validation - check window functions.
+            QueryModel m = queryModel;
+            while (m != null) {
```

Comment:
nit: we could merge this loop with the below one. It's fine if both start at `queryModel`.